lib/smppex/mc.ex

```diff
@@ -515,10 +532,30 @@ defmodule SMPPEX.MC do
     {:reply, :ok, new_st}
   end
 
-  defp do_handle_stop(st) do
-    _ = st.module.handle_stop(st.module_state)
+  defp do_handle_stop(reason, st) do
+    lost_pdus = PduStorage.fetch_all(st.pdus)
     :ok = PduStorage.stop(st.pdus)
-    {:stop, :normal, :ok, st}
+
+    # TODO: remove legacy implementation handling in future versions
```

Comment:
To avoid legacy polyfilling, how about renaming the function to `terminate(reason, lost, state)` to match `GenServer`? The old function would just retain it's signature, but emit a deprecation warning (`handle_stop/1` is deprecated in favour of `terminate/3`)